src/main/navigation/navigation_fixedwing.c

```diff
@@ -463,17 +463,20 @@ int16_t applyFixedWingMinSpeedController(timeUs_t currentTimeUs)
 
 int16_t fixedWingPitchToThrottleCorrection(int16_t pitch)
 {
-    if (pitch > navConfig()->fw.pitch_to_throttle_thresh) {
-        // Positive pitch above threshold
-        return DECIDEGREES_TO_DEGREES(pitch - navConfig()->fw.pitch_to_throttle_thresh) * navConfig()->fw.pitch_to_throttle;
+    // Calculate base throttle correction from pitch moving average
+    const int16_t movingAverageCycles = 128; //Number of main loop cycles for average calculation
+    static int16_t averagePitch = (averagePitch * movingAverageCycles + pitch - averagePitch) / movingAverageCycles;
+    const int16_t baseThrottleCorrection = DECIDEGREES_TO_DEGREES(averagePitch) * navConfig()->fw.pitch_to_throttle;
```

Comment:
This line already accomplishes what I had in mind for using a smoothed pitch angle for the throttle correction input.